src/wasm/asm-wasm-builder.cc

```diff
@@ -453,26 +574,108 @@ class AsmWasmBuilderImpl : public AstVisitor {
   }                                                                   \
   break
 
+  Expression* GetLeft(BinaryOperation* expr) {
```

Comment:
Why are we assuming that the non-trivial expression is on the left hand side in so many places?